modules/decision/search_pattern.py

```diff
@@ -0,0 +1,242 @@
+        """
+        # Compute the size of square. First square will be a distance of the minimum of width and
+        # depth, each subsequent square will be a width larger
+        square_size = (
+            min(self.__search_gap_depth, self.__search_gap_width)
+            + (self.__current_square - 1) * self.__search_gap_width
+        )
+
+        # If the depth is less than the width, we apply an offset at each corner to ensure the
+        # entire area is scanned
+        adjustment = 0
+        if self.__search_gap_depth < self.__search_width:
+            adjustment = (self.__search_gap_width - self.__search_gap_depth) / 2
+
+        # Calculate the corners based on the offsets and the search origin. Top left corner is moved
+        # right by search_gap_width as the final side of the square will instead cover that part
+        self.__square_corners = [
+            (
+                self.__search_origin_x
+                - square_size
+                - adjustment
+                + self.__search_gap_width,
+                self.__search_origin_y + square_size,
+            ),  # Top left corner
+            (
+                self.__search_origin_x + square_size,
+                self.__search_origin_y + square_size + adjustment,
+            ),  # Top right corner
+            (
+                self.__search_origin_x + square_size + adjustment,
+                self.__search_origin_y - square_size,
+            ),  # Bottom right corner
+            (
+                self.__search_origin_x - square_size,
+                self.__search_origin_y - square_size - adjustment,
+            ),  # Bottom left corner
+        ]
+
+    def __calculate_side_of_square(self):
+        """
+        Computes the gaps along the current side of the square
+        """
+        # Calculate the positions of the current and next corners
+        self.__current_corner = self.__square_corners[self.__current_side_in_square]
+        next_corner = self.__square_corners[(self.__current_side_in_square + 1) % 4]
+
+        # Determine if the drone is moving horizontally or vertically along the current side
+        self.__moving_horizontally = self.__current_side_in_square % 2 == 0
+
+        # Calculate the length of the current side based on the direction of movement
+        # Note that this is signed (e.g. travellingin -x direction has a negative sign)
+        if self.__moving_horizontally:
+            side_length = next_corner[0] - self.__current_corner[0]
+        else:
+            side_length = next_corner[1] - self.__current_corner[1]
+
+        # As calculated, it is the distance to the next corner, however, we want to stop before
+        # we actually reach the next side as the area will be covered by it
+        if side_length > 0:
+            side_length += (self.__search_gap_depth - self.__search_gap_width) / 2
+        else:
+            side_length -= (self.__search_gap_depth - self.__search_gap_width) / 2
```

Comment:
why do we subtract here, also, if this is the very first case and is never used again, we don't want to evaluate the if every time, so let's cover it in the init (maybe by setting it to this value in the first place

Reply:
This is because the way I'm currently doing it the side length is signed (i.e. positive when moving up/right or negative when moving down/left).